### PR TITLE
DevTools > Accessibility > Contrast issues

### LINKED ISSUE #3266: [devtools-tips] 05 - Discover and fix low contrast text
Details: b/205837080
Deadline: Oct 6
Notes: Do not merge the PR when ready, tag @jecfish to merge in time with video publish
Also refresh relevant docs, and include the video

I think these info here worth a page of documentation of itself

### PR DESCRIPTION
**Pull request by sofiayem**
Rewrote the Color Picker section in the a11y doc.

Now it's CUJ-based with the following goal and steps:
Goal: Make your website more readable
Steps: (1) Discover low contrast text. (2) Fix low contrast text.

Additionally, backfilled a promo blog post for DTT 5: Discover and fix low contrast text with DevTools.

Fixes #3266

### CODE REVIEW

File: site/en/blog/devtools-tips-5/index.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/G1vFeA4jqqbuXHnW6IZK.png", alt="Contrast issues in CSS Overview and Issues panels.", width="800", height="644" %}
+
+- Emulate vision deficiencies to understand what your page looks like for all your users.
+
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/QdE5Gr23r68leiaVjfUI.png", alt="Emulating blurred vision.", width="800", height="500" %}
+
+To learn more, see:
+
+- [Color and contrast accessibility](https://web.dev/color-and-contrast-accessibility/)
+- [WebAIM guidelines](https://webaim.org/standards/wcag/)
```
Comment (heyawhite) [inline site/en/blog/devtools-tips-5/index.md:46]:
```suggestion
- [Color and contrast accessibility](https://web.dev/color-and-contrast-accessibility/)
- [WebAIM guidelines](https://webaim.org/standards/wcag/)
- [Learn Accessibility's color and contrast module](https://web.dev/learn/accessibility/color-contrast/)
```

Comment (heyawhite) [reply site/en/blog/devtools-tips-5/index.md:46]: I'd love to draw attention to this new resource in these docs, if you're ammendable!

File: site/en/blog/devtools-tips-5/index.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/tolxw3v0DlcOHGZhbJPZ.png", alt="Recommended contrast ratio lines.", width="800", height="525" %}
+
+- Discover all contrast issues in the **CSS Overview** and (preview) **Issues** panels.
+
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/G1vFeA4jqqbuXHnW6IZK.png", alt="Contrast issues in CSS Overview and Issues panels.", width="800", height="644" %}
+
+- Emulate vision deficiencies to understand what your page looks like for all your users.
+
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/QdE5Gr23r68leiaVjfUI.png", alt="Emulating blurred vision.", width="800", height="500" %}
```
Comment (heyawhite) [inline site/en/blog/devtools-tips-5/index.md:41]: Can you add more detail to the alt text for these images? For example:

```
  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/odkqE1ed3sgVxnAypWwF.png", alt="Contrast ratios are available in a tooltip, with a Color Picker to measure the ration of alternative colors. AA and AAA grading of the ration is available.", width="800", height="525" %}
```

**Review by heyawhite — COMMENTED** [review]
Excited about these blog posts! I know you're in the middle, and I'm not acting as a full reviewer. I wanted to offer the new Learn A11y module as a resource (and, because I'm me, a suggestion on alt text).

Hope that it's useful :)

File: site/en/blog/devtools-tips-5/index.md (hunk 1 above)
Comment (sofiayem) [reply site/en/blog/devtools-tips-5/index.md:46]: Sure

**Comment by sofiayem** [comment]
> Excited about these blog posts! I know you're in the middle, and I'm not acting as a full reviewer. I wanted to offer the new Learn A11y module as a resource (and, because I'm me, a suggestion on alt text).
> 
> Hope that it's useful :)

Thanks for your suggestions. Yeah, I'm in the middle and I plan to rework the entire a11y section in docs (https://developer.chrome.com/docs/devtools/accessibility/reference/) over several future PRs.

File: site/en/blog/devtools-tips-5/index.md (hunk 2 above)
Comment (sofiayem) [reply site/en/blog/devtools-tips-5/index.md]: Thank you. Applied.

File: site/en/docs/devtools/accessibility/reference/index.md [hunk 3]
```diff
@@ -185,50 +171,95 @@ View an element's computed accessibility properties in [the Accessibility pane][
 
 {% Img src="image/admin/KhrNElIocLRywp4Na7x7.png", alt="The Computed (Accessibility) Properties section.", width="800", height="506" %}
 
-**Figure 9**. The Computed (Accessibility) Properties section
-
-## View the contrast ratio of a text element in the Color Picker {: #contrast }
+## Make your website more readable {: #contrast }
```
Comment (jecfish) [inline site/en/docs/devtools/accessibility/reference/index.md:174]: Should we move this part to a documentation itself (e.g. `[devtools/accessibility/fix-low-contrast)`?

File: site/en/blog/devtools-tips-5/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- Emulate vision deficiencies to understand what your page looks like for all your users.
+
+  {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/QdE5Gr23r68leiaVjfUI.png", alt="Blurred vision selected in the Rendering tab and emulated in the viewport.", width="800", height="500" %}
+
+For a more hands-on learning experience, see the [Discover and fix low contrast text](/docs/devtools/accessibility/reference/#contrast) tutorial.
+
+To learn more, see:
+
+- [Color and contrast accessibility](https://web.dev/color-and-contrast-accessibility/)
+- [WebAIM guidelines](https://webaim.org/standards/wcag/)
```
Comment (jecfish) [inline site/en/blog/devtools-tips-5/index.md:48]: Add a link to `https://web.dev/learn/accessibility/` as a learning resource!

File: site/en/docs/devtools/accessibility/reference/index.md (hunk 3 above)
Comment (sofiayem) [reply site/en/docs/devtools/accessibility/reference/index.md:174]: Yeah, I have a few ideas about that. I'll explain them in an email soonish.

File: site/en/docs/devtools/accessibility/reference/index.md [hunk 5]
```diff
@@ -5,10 +5,11 @@ authors:
   - kaycebasques
   - sofiayem
 date: 2022-06-09
-#updated: YYYY-MM-DD
+updated: 2022-10-12
 description: "A comprehensive reference of accessibility features in Chrome DevTools."
 tags:
   - accessibility
+  - find-issues
 ---
```
[the remark this answers is not in the document]
Comment (sofiayem) [reply site/en/docs/devtools/accessibility/reference/index.md:14]: Added.

**Comment by sofiayem** [comment]
Decided to create a dedicated doc for "How to fix contrast" and separate it from the features reference. Not merging yet.

File: site/en/docs/devtools/accessibility/contrast/index.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/Jth5CZ7BXPUKajXhmgFR.png", alt="The tooltip shows a warning sign next to the contrast value.", width="800", height="507" %}
+
+   Notice the {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/xk7aqSvZfY5vU66QtCer.svg", alt="Warning.", width="20", height="20" %} warning sign next to the contrast ratio value in the tooltip. The contrast ratio measures the difference in brightness between the foreground (text color) and background colors.
+
+1. [Open the **Color Picker**](/docs/devtools/css/reference/#color-picker) next to the color declaration of the element's text and, in the **Color Picker**, expand the **Contrast ratio** section.
+
+   {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/uiUWw7dczJ6VonjO5Gl3.png", alt="The Contrast ratio section of the Color Picker.", width="800", height="598" %}
+
+   The **Color Picker** tells you the contrast ratio doesn't meet neither AA nor AAA levels of [WebAIM guidelines](https://webaim.org/standards/wcag/).
```
Comment (rachelandrew) [inline site/en/docs/devtools/accessibility/contrast/index.md:106]:
```suggestion
   The **Color Picker** tells you the contrast ratio doesn't meet neither AA nor AAA levels of [WebAIM guidelines](https://webaim.org/standards/wcag/).
```
```suggestion
   The **Color Picker** tells you the contrast ratio doesn't meet the AA or AAA levels of [WebAIM guidelines](https://webaim.org/standards/wcag/).
```

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**